internal/cli/atlas/privateendpoints/gcp/watch.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type WatchOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	id    string
+	store store.PrivateEndpointDescriber
+}
+
+func (opts *WatchOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.PrivateEndpoint(opts.ConfigProjectID(), provider, opts.id)
+	if err != nil {
+		return false, err
+	}
+	return result.Status == "AVAILABLE" || result.Status == "FAILED", nil
+}
+
+func (opts *WatchOpts) Run() error {
+	if err := opts.Watch(opts.watcher); err != nil {
+		return err
+	}
+
+	return opts.Print(nil)
+}
+
+// mongocli atlas privateEndpoints gcp watch <privateEndpointId> [--projectId projectId].
+func WatchBuilder() *cobra.Command {
+	opts := &WatchOpts{}
+	cmd := &cobra.Command{
+		Use:   "watch <privateEndpointId>",
+		Short: "Watch for a specific GCP private endpoint to be available.",
```

Comment:
@blva Bianca, how about: "Watch the specified GCP private endpoint to detect changes".
Question: why did you add "to be available"? What happens if I run this command? Does it set a watcher and if so, for how long does the watcher last? Does it notify me of changes? Does it watch ONLY for the availability of the endpoint? How does running this differ from a ping? What's the suggested usage for users?

Reply:
**Q: Why did you add "to be available"? What happens if I run this command?**

This command will keep checking the endpoint status periodically until it reaches an AVAILABLE or FAILED state.

 **Q: Does it set a watcher and if so, for how long does the watcher last** (note: Double checking with @andreaangiolillo )
The watcher will keep watching until the expected status is reached. Otherwise, user can interrupt the watcher at any time if the command is ran in a terminal.

**Q: Does it notify me of changes?** 
It prints `GCP Private endpoint changes completed.` once expected statuses are reached.

**Q: Does it watch ONLY for the availability of the endpoint?** 
Based on the previous commands, we do check for availability but also if it fails we stop to watch. I'd say it checks for terminal states. 

**Q: How does running this differ from a ping?**
We're polling the status of the endpoint instead of waiting for it to respond to us. The command will be constantly getting an answer but we'll be waiting for the right status.

**Suggestion: "Watch the specified GCP private endpoint to detect changes".** 

Sounds good to me. Please let me know if you change your mind given the extra info added.

Reply:
@blva let's do this:
1. For "Short", use:

Short: `Watch the specified GCP private endpoint to detect changes in the endpoint's state.`,

2. Add "Long" with this description:
Long: `This command checks the endpoint's state periodically until the endpoint reaches an AVAILABLE or FAILED state. Once the endpoint reaches the expected state, the command prints "GCP Private endpoint changes completed."  If you run the command in the terminal, it blocks the terminal session until the resource becomes available or fails. You can interrupt the watcher polling at any time.`,

Reply:
SGTM. Maybe we can add 

> You can interrupt the command's polling at any time with CTRL-C

Reply:
thanks, will add that as well!

Reply:
SGTM (Also love the discussion here, good job on both sides refining the Long description)

Reply:
Thanks Colm. Good question, I've checked other watch commands and so far we have been adding only the `Short` description. This will be the first watch command we're adding a Long description. If we agree, we can come up with proposals for the other watch commands as they might need more clarity similarly to this one. If we agree on that I can open a new PR w/ docs update for watch commands. 

From my side it sounds like a great idea, however I wonder if we want to add Long description to all commands then or if we're OK w/ having only one type of command with longer descriptions. WDYT?

Reply:
Thank, good to know. I've created a [CLOUDP-110093](https://jira.mongodb.org/browse/CLOUDP-110093) for tracking this change and will proceed with adding the Long description for this specific command. In `CLOUDP-110093` I'll work on the base long string to use across all other watch commands.